src/idea.js

```diff
@@ -1,10 +1,11 @@
 class Idea {
-  constructor(title, body) {
-    this.id = Date.now();
+  constructor(title, body, id) {
+    this.id = id || Date.now();
```

Comment:
Having the data model check for an existing id rather then making a new id every single time the page was generated was awesome to see!